### PR TITLE
Net: Divide ProcessMessage in smaller functions

### PR DESCRIPTION
**Pull request by jtimon**
As in #9579, indentation and further documentation can be done later to avoid further disruption on the same PR. Maybe at different times for different groups of functions.
The main goal of this PR ProcessMessage to make it more readable and maintainable (ie, changes to ProcessMessage should in principle be easier to review after this PR). Brings ProcessMessage to 142 lines.

This should be relatively easy to review.

### CODE REVIEW

**Comment by gmaxwell** [comment]
This should use a switch statement, not a long if/else cascade.

**Comment by sipa** [comment]
You can't switch/case on strings.

**Comment by gmaxwell** [comment]
Well damn, I thought it was an enum now.

**Comment by gmaxwell** [comment]
I feel like this just obscures the control flow. :-/

**Comment by rebroad** [comment]
What problem is this PR fixing?

**Comment by dcousens** [comment]
@rebroad algorithmic readability,  it is simpler to reason about these algorithms,  IMHO,  by these smaller functions than it is otherwise.

**Comment by jtimon** [comment]
> What problem is this PR fixing?

Right, readability and maintainability (ie ease of review for later changes on ProcessMessage). As noted, this can be indented later (although that obscures the git blame).
In other projects, having to read more than X nested control flows in function directly goes against the style and smaller functions can be always a solution to that. Others simply don't allow functions above N lines

IMO, before this PR ProcessMessage is both too long and has some lines with too many nestings. It is very hard for me to read this function as it, and I suspect I'm not the only one.

> I feel like this just obscures the control flow. :-/

Yeah, how so?

**Comment by sipa** [comment]
> (although that obscures the git blame).

Use `git blame -w`.

File: src/net_processing.cpp [hunk 1]
```diff
@@ -2558,20 +2545,149 @@ bool static ProcessMessage(CNode* pfrom, string strCommand, CDataStream& vRecv,
...
+    else if (strCommand == NetMsgType::PING)
+    {
+        return MsgPing(pfrom, vRecv, connman, msgMaker);
+    }
+    else if (strCommand == NetMsgType::PONG)
+    {
+        return MsgPong(pfrom, vRecv, nTimeReceived);
+    }
+    else if (strCommand == NetMsgType::FILTERLOAD)
+    {
```
Comment (MarcoFalke) [inline src/net_processing.cpp]: According to our style guideline all those  curly brackets are on the "wrong" line. If we want to stick to the current style guideline, new code should not violate it. If we don't want to stick with the current style guide, we should weaken in with regard to new lines.

**Review by maflcko — COMMENTED** [review]
Concept ACK

File: src/net_processing.cpp (hunk 1 above)
Comment (jtimon) [inline src/net_processing.cpp]: You are completely right, I'm happy correcting this.

**Comment by jtimon** [comment]
Added a couple of commits to hopefully squash.

**Comment by gmaxwell** [comment]
How does it obscure the control flow? Because there is only one caller to these functions, and they're all called in order. But now looking at the function to considering its invariants you can no longer see where they are called from, what threads they run, what orders things happen in...  As far as nesting goes... it hardly changes the nesting-- just reduces the indentation by one.  It doesn't change the cyclomatic complexity of the code (which is usually the motivation against large functions, avoiding gnarly intertwined functions that are not testable).

I don't think it grievously hurts things, but I do not see the improvement and it feels a bit like movement for movement sake to me. No biggie.

**Comment by jtimon** [comment]
Needed rebase. Rebased on top of #9659 some changes to use more const can be made even if the separation is not done.

**Comment by jnewbery** [comment]
Concept ACK, as long as you chance the function names to `ProcessMsg[type]()`. You're proposing to refactor `SendMessages()` in #9579. I'd prefer that function to be split into `SendMsg[type]()` in that PR if you continue. At the moment you have a function here `MsgFeefilter()` for processing a received feefilter message and a function in #9579 called `MsgFeefilter()` for sending a feefilter message. That's not helpful for code taging and documentation.

**Comment by jtimon** [comment]
Fixed nits ( https://github.com/bitcoin/bitcoin/pull/9608#issuecomment-283545547 https://github.com/bitcoin/bitcoin/pull/9579#discussion_r103590711 ) and rebased without needing it.

**Review by dcousens — APPROVED** [review]

**Comment by dcousens** [comment]
once over utACK 1d7a16f

**Comment by theuni** [comment]
Concept ACK, but I don't see this as an improvement without separating control flow and dispatch. I believe that's part of @gmaxwell's concern as well?

Added that here: https://github.com/theuni/bitcoin/commit/f1e4e281e3f1eb884f8010ac941c82752174bdbe (needs tests). This has the advantage of moving the rules for message acceptance into one place. If the message is deemed suitable, it can be dispatched with no ordering/dependency concerns.

**Comment by jtimon** [comment]
@theuni At a first glance that looks like an improvement in readability. I'm happy with that being done before or after this, probably before is less disruptive.

**Comment by jtimon** [comment]
Needed rebase.

**Comment by JeremyRubin** [comment]
@jtimon can you take a look at #10145 as it now does a similar thing.

**Comment by jtimon** [comment]
Needed rebase

**Comment by jtimon** [comment]
Needed rebase.

**Comment by jtimon** [comment]
I can reopen and rebase if there's interest, but it seems there has been no interest for a while, so closing.

**Pull request closed without merging**